#!/usr/bin/env python
PACKAGE = "orp"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

##############################################################################

#       name             type      1  description                        default min  max
# RECOGNITION
gen.add("save_pcd",                            bool_t, 0, "", False)
gen.add("save_cph",                            bool_t, 0, "", False)
gen.add("save_vfh",                            bool_t, 0, "", False)
gen.add("save_cvfh",                           bool_t, 0, "", False)
gen.add("save_sixdof",                         bool_t, 0, "", True)

gen.add("vfh_radius_search",                            double_t, 0, "", 0.03, 0.001, 0.1)
gen.add("cvfh_radius_search",                            double_t, 0, "", 0.03, 0.001, 0.1)
gen.add("cph_vertical_bins",                  int_t,    0, "number of vertical slices for CPH histogram", 5,1,50)
gen.add("cph_radial_bins",                  int_t,    0, "number of radial slices for CPH histogram", 72, 1, 360)

##############################################################################

exit(gen.generate(PACKAGE, "orp", "HistogramSaver"))
